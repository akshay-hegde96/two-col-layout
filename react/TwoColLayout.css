.layoutContainer{
  padding: 25px;
}
.row {
  display: flex;
}

.column {
  flex: 50%;
}
.heading{
  display: flex;
  justify-content: center;
}
.subHeading{
  display: flex;
  justify-content: center;
}

/*==== mobile ===== */
@media only screen and (max-width:480px){
  .row {
    display: block;
  }
}
