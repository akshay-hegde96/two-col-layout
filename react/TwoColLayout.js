import React from 'react'
import styles from './TwoColLayout.css'
import { useCssHandles } from 'vtex.css-handles'

const CSS_HANDLES = [
  'image1',
  'image2',
  'image3',
  'image4',
  'imagesWrapper',
  'firstColParaContainer',
  'firstColFirstParagraph',
  'firstColSecondParagraph',
  'firstColThirdParagraph',
  'firstColFourthParagraph',
  'secondColFirstParagraph',
  'secondColSecondParagraph',
  'firstColThirdParagraph',
  'secondColFourthParagraph',
]

const TwoColLayout = (props) => {
  const handles = useCssHandles(CSS_HANDLES)
  return (
    <div className={styles.layoutContainer}>
      <h2 className={styles.heading}>{props.heading}</h2>
      <h4 className={styles.subHeading}>{props.subHeading}</h4>
      <div className={styles.row}>
        <div className={styles.column}>
          {props.imageUrl1 ||
          props.imageUrl2 ||
          props.imageUrl3 ||
          props.imageUrl4 ? (
            <div className={handles.imagesWrapper}>
              <img src={props.imageUrl1} className={handles.image1} />
              <img src={props.imageUrl2} className={handles.image2} />
              <img src={props.imageUrl3} className={handles.image3} />
              <img src={props.imageUrl4} className={handles.image4} />
            </div>
          ) : (
            <div className={handles.firstColParaContainer}>
              <p className={handles.firstColFirstParagraph}>
                {props.firstColFirstParagraph}
              </p>
              <p className={handles.firstColSecondParagraph}>
                {props.firstColSecondParagraph}
              </p>
              <p className={handles.firstColThirdParagraph}>
                {props.firstColThirdParagraph}
              </p>
              <p className={handles.firstColFourthParagraph}>
                {props.firstColFourthParagraph}
              </p>
            </div>
          )}
        </div>
        <div className={styles.column}>
          <p className={handles.secondColFirstParagraph}>
            {props.secondColFirstParagraph}
          </p>
          <p className={handles.secondColSecondParagraph}>
            {props.secondColSecondParagraph}
          </p>
          <p className={handles.secondColThirdParagraph}>
            {props.secondColThirdParagraph}
          </p>
          <p className={handles.secondColFourthParagraph}>
            {props.secondColFourthParagraph}
          </p>
        </div>
      </div>
    </div>
  )
}

TwoColLayout.schema = {
  title: 'TwoColLayout',
  description: 'Edit the required section of TwoColLayout page.',
  type: 'object',
  properties: {
    heading: {
      title: 'Heading',
      description: 'Page Heading',
      type: 'string',
      default: null,
    },
    subHeading: {
      title: 'Sub-Heading',
      description: 'sub heading for the page',
      type: 'string',
      default: null,
    },
    imageUrl1: {
      title: 'Image URL one',
      description: 'URL of the image to be displayed on left side.',
      type: 'string',
      default: null,
    },
    secondColFirstParagraph: {
      title: 'Paragraph1',
      description: 'Right side First Paragraph.',
      type: 'string',
      default: null,
    },
    secondColSecondParagraph: {
      title: 'Paragraph2',
      description: 'Right side Second Paragraph.',
      type: 'string',
      default: null,
    },
  },
}
export default TwoColLayout
