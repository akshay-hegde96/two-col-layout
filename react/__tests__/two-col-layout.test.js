import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import TwoColLayout from '../TwoColLayout'

configure({ adapter: new Adapter() })

describe('TwoColLayout', () => {
  it('should show heading text', () => {
    const wrapper = shallow(<TwoColLayout heading="This is heading" />)
    const text = wrapper.find('div h2')

    expect(text.text()).toBe('This is heading')
  })
  it('should show sub-heading text', () => {
    const wrapper = shallow(
      <TwoColLayout
        heading="This is heading"
        subHeading="this is sub-heading"
      />
    )

    const text = wrapper.find('div h4')

    expect(text.text()).toBe('this is sub-heading')
  })
  it('should render correctly', () => {
    const component = shallow(<TwoColLayout />)

    expect(component).toMatchSnapshot()
  })
})
