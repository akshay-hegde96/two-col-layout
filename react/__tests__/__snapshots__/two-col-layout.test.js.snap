// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoColLayout should render correctly 1`] = `ShallowWrapper {}`;
